tgui/packages/tgui/interfaces/LightController.tsx

```diff
@@ -0,0 +1,319 @@
+                  (id) => (
+                    <Tabs.Tab
+                      key={id}
+                      selected={currentTemplate === id}
+                      onClick={() => setCurrentTemplate(id)}>
+                      <Box fontSize="14px" textColor={"#cee"}>
+                        {templates[id].light_info.name}
+                      </Box>
+                      <Box ml={0.1} className={classes(['lights32x32', "light_fantastic_" + id])} />
+                    </Tabs.Tab>
+                  ))}
+              </Tabs>
+            </Section>
+          </Stack.Item>
+          <Stack.Item grow>
+            <LightControl info={light_info} />
+            <LightInfo light={templates[currentTemplate]} />
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+type LightControlProps = {
+  info : LightDetails;
+};
+
+const LightControl = (props : LightControlProps, context) => {
+  const { act, data } = useBackend<Data>(context);
+  const { on } = data;
+  const { info } = props;
+  return (
+    <Section>
+      <Stack vertical justify="space-between" fill>
+        <Stack.Item>
+          <Stack justify="space-between">
+            <Stack.Item>
+              <Box fontSize="16px" mt={0.5}>
+                {info.name}
+              </Box>
+            </Stack.Item>
+            <Stack.Item>
+              <Button
+                fontSize="16px"
+                icon="brush"
+                textColor={info.color}
+                onClick={() => act("change_color")}
+                content={info.color}
+                />
+              <Button
+                fontSize="16px"
+                color={on ? "good" : "bad"}
+                icon="power-off"
+                onClick={() => act("set_on", {
+                value: !on,
+                })} />
+            </Stack.Item>
+          </Stack>
+        </Stack.Item>
+        <Stack.Item>
+          <Stack justify="space-around">
+            <Stack.Item>
+              <Section title="Direction" textAlign="center" fontSize="11px">
+                <DirectionSelect />
+              </Section>
+            </Stack.Item>
+            <Stack.Item>
+              <Section title="Angle" textAlign="center" fontSize="11px">
+                <AngleSelect />
+              </Section>
+            </Stack.Item>
+          </Stack>
+        </Stack.Item>
+        <Stack.Item align="end">
+          <Slider
+            unit="tiles"
+            value={info.range}
+            color="blue"
+            minValue={0}
+            maxValue={10}
+            onChange={(e, value) => act("set_range", {
+              value: value,
+            })}
+            step={0.1}
+            stepPixelSize={5}
+            />
+          <Slider
+            unit="intensity"
+            value={info.power}
+            color="olive"
+            minValue={-1}
+            maxValue={5}
+            format={(value) => { return round(value, 2); }}
```

Comment:
do you need to wrap this in curly braces with a return here? should be able to just value => round(

Reply:
yep, you're right